zero_bin/rpc/src/lib.rs

```diff
@@ -135,6 +129,85 @@ where
             }
         });
 
+    Ok(prev_hashes)
+}
+
+async fn fetch_previous_block_hashes_smart_contract<ProviderT, TransportT>(
+    cached_provider: Arc<CachedProvider<ProviderT, TransportT>>,
+    target_block_number: u64,
+) -> anyhow::Result<PreviousBlockHashes>
+where
+    ProviderT: Provider<TransportT>,
+    TransportT: Transport + Clone,
+{
+    let bytes = cached_provider
+        .as_provider()
+        .raw_request::<_, Bytes>(
+            "eth_call".into(),
+            (json!({"input": "0x60005B60010180430340816020025280610101116300000002576120205FF3"}), target_block_number),
+        )
+        .await?;
+
+    let prev_hashes = bytes
+        .chunks(32)
+        .skip(1) // blockhash for current block
+        .map(FixedBytes::<32>::try_from)
+        .rev()
+        .collect::<Result<Vec<_>, _>>()?;
+
+    PreviousBlockHashes::try_from(prev_hashes)
+        .map_err(|_| anyhow!("Invalid number of block hashes"))
```

Comment:
`PreviousBlockHashes`  is array of 256 elements, so it would fail is `prev_hashes` length is not 256(happened before).  We always have chunks of 32 bytes converting to `FixedBytes<32>` - that could fail only for the last chunk being smaller than 32. I will update error message for that case.